### PR TITLE
Weight sharing dim

### PR DESCRIPTION
**Pull request by wiseodd**
Basically just reverting to the old one.

@runame please review.

@kazukiosawa I would appreciate it if you could merge & update PyPi! Thanks!

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,5 @@
...
-requires = [
-    "setuptools>=42",
-    "wheel"
-]
-build-backend = "setuptools.build_meta"
\ No newline at end of file
+requires = ["setuptools>=42", "wheel"]
+build-backend = "setuptools.build_meta"
+[tool.ruff.format]
+quote-style = "single"
```
Comment (runame) [inline pyproject.toml:5]: Is there a reason to add ruff here? I assume the whole code base (besides the file modified in this PR) is not formatted with ruff.

File: asdl/gradient.py [hunk 2]
```diff
@@ -50,27 +51,22 @@ def data_loader_gradient(
...
         grads = []
         for module in model.modules():
             g = cxt.batch_grads(module, flatten=True)
             if g is not None:
-                if batch_size is not None and batch_size != g.shape[0]:
-                    # Reduce the weight-sharing dim
-                    g = g.reshape(batch_size, -1, g.shape[-1]).sum(1)
                 grads.append(g)
-        grads = torch.cat(grads, dim=1)  # (n, p)
+        grads = torch.cat(grads, dim=1)  # (..., p)
```
Comment (runame) [inline asdl/gradient.py:69]: Shouldn't this be `dim=-1`?

**Review by runame — CHANGES_REQUESTED** [review]

**Comment by wiseodd** [comment]
@runame could you check once more. The requested changes should be resolved.

**Review by runame — APPROVED** [review]
Thanks, LGTM.

**Review by kazukiosawa — APPROVED** [review]
Thank you for the PR! LGTM.
